src/main/java/software/amazon/awssdk/crt/io/Uri.java

```diff
@@ -0,0 +1,49 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+package software.amazon.awssdk.crt.io;
+
+import java.nio.charset.StandardCharsets;
+import software.amazon.awssdk.crt.CRT;
+
+public class Uri {
+    static {
+        new CRT();
+    };
+
+    public static String appendEncodingUriPath(String encoded, String path) {
+        return new String(
+                appendEncodingUriPath(encoded.getBytes(StandardCharsets.UTF_8), path.getBytes(StandardCharsets.UTF_8)),
+                StandardCharsets.UTF_8);
+    }
+
+    public static String appendEncodingUriPath(String path) {
+        return appendEncodingUriPath("", path);
+    }
+
+    public static String appendEncodingUriParam(String encoded, String param) {
+        return new String(appendEncodingUriParam(encoded.getBytes(StandardCharsets.UTF_8),
+                param.getBytes(StandardCharsets.UTF_8)), StandardCharsets.UTF_8);
```

Comment:
Because looking around the code base, I saw byte arrays passed all over the place, and strings passed only in a few places (Log, AwsSigning, TlsContextOptions, MQTTClientConnectionOptions) so I decided to do what we do most other places. One way or another we need to convert from jstring to bytebuffer/cursor, I guess I could have used 
`(*env)->GetStringUTFChars`